MeshLib/MeshEditing/MeshRevision.cpp

```diff
@@ -67,39 +67,68 @@ MeshLib::Mesh* MeshRevision::simplifyMesh(const std::string &new_mesh_name, doub
 	if (this->_mesh.getNElements() == 0)
 		return nullptr;
 
+	// data structures for the new mesh
 	std::vector<MeshLib::Node*> new_nodes = this->constructNewNodesArray(this->collapseNodeIndices(eps));
 	std::vector<MeshLib::Element*> new_elements;
-
-	const std::vector<MeshLib::Element*> &elements(this->_mesh.getElements());
-	for (auto elem = elements.begin(); elem != elements.end(); ++elem)
-	{
-		unsigned n_unique_nodes(this->getNUniqueNodes(*elem));
-		if (n_unique_nodes == (*elem)->getNBaseNodes() && (*elem)->getDimension() >= min_elem_dim)
+	MeshLib::Properties new_properties;
+	boost::optional<PropertyVector<int> &> new_material_vec(
+		new_properties.createNewPropertyVector<int>(
+			"MaterialIDs", MeshItemType::Cell, 1
+		)
+	);
+
+	// original data
+	std::vector<MeshLib::Element*> const& elements(this->_mesh.getElements());
+	MeshLib::Properties const& properties(_mesh.getProperties());
+	boost::optional<MeshLib::PropertyVector<unsigned> const&> material_vec(
+		properties.getPropertyVector<unsigned>("MaterialIDs"));
```

Comment:
>i think something is wrong here with the sequence of commands. you should first check if the old mesh had a "material id" property and only if it did should you create such a property for the new mesh.

I'll change this.

> additional comment: it might not be relevant now, but in theory you would have to copy all properties from the old mesh. i suppose we need some methods for the properties-object to be able to do this in the future ....

Yes, that is true. This PR is one of the first steps. After that there is still much to do.